### PR TITLE
[Sync] Sync 1.1.7

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR sync `master` with `release/1.2-dev` branch.

Fixes # (issue) <- this [links related issue to this PR](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by justusschock — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,19 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
+## [unreleased] - YYYY-MM-DD
```
Comment (carmocca) [inline CHANGELOG.md:7]: This is duplicated

**Comment by Borda** [comment]
can we squash 96dff0885dfbc23a49f1549a7f417d44b11fa3c2 and 96dff0885dfbc23a49f1549a7f417d44b11fa3c2

File: dockers/tpu-tests/tpu_test_cases.jsonnet [hunk 2]
```diff
@@ -22,9 +22,9 @@ local tputests = base.BaseTest {
     |||
       cd pytorch-lightning
       coverage run --source=pytorch_lightning -m pytest -v \
-          pytorch_lightning/utilities/xla_device_utils.py \
-          tests/accelerators/legacy/test_tpu_backend.py \
-          tests/models/test_tpu.py
+        pytorch_lightning/utilities/xla_device_utils.py
```
Comment (rohitgr7) [inline dockers/tpu-tests/tpu_test_cases.jsonnet:25]:
```suggestion
        pytorch_lightning/utilities/xla_device_utils.py \
```

File: dockers/tpu-tests/tpu_test_cases.jsonnet [hunk 3]
```diff
@@ -22,9 +22,9 @@ local tputests = base.BaseTest {
     |||
       cd pytorch-lightning
       coverage run --source=pytorch_lightning -m pytest -v \
-          pytorch_lightning/utilities/xla_device_utils.py \
-          tests/accelerators/legacy/test_tpu_backend.py \
-          tests/models/test_tpu.py
+        pytorch_lightning/utilities/xla_device_utils.py \
+        tests/accelerators/legacy/test_tpu_backend.py \
+        tests/models/test_tpu.py
```
Comment (rohitgr7) [inline dockers/tpu-tests/tpu_test_cases.jsonnet:27]:
```suggestion
          pytorch_lightning/utilities/xla_device_utils.py \
          tests/accelerators/legacy/test_tpu_backend.py \
          tests/models/test_tpu.py
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -172,6 +185,30 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+
+
+### Changed
+
+
+### Deprecated
+
+
+### Removed
+
```
Comment (Borda) [inline CHANGELOG.md:200]:
```suggestion
```

**Review by Borda — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,19 @@ All notable changes to this project will be documented in this file.
...
+### Added
+
+### Changed
+
+### Deprecated
+
+### Removed
+
+### Fixed
```
Comment (Borda) [inline CHANGELOG.md:17]:
```suggestion
```

**Review by rohitgr7 — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Pull request merged**